intents.go

```diff
@@ -27,111 +44,122 @@ const (
 	Expired      ProcessingStatus = "Expired"
 	OnChain      ProcessingStatus = "OnChain"
 	Invalid      ProcessingStatus = "Invalid"
-	// evaluate adding specific on chain statuses
 )
 
+// Asset structure for TokenFrom, CurrencyTo, and ReStakeTo
+type Asset struct {
+	Type    Type     `json:"Type"`
+	Address string   `json:"Address"`
+	Amount  string   `json:"Amount"`
+	ChainId *big.Int `json:"ChainId"`
+}
+
+// Intent represents a user's intention to perform a transaction, which could be a swap, buy, sell,
+// staking, or restaking action.
 type Intent struct {
-	Sender            string           `json:"sender" binding:"required,eth_addr"` // filled by ui
-	Kind              string           `json:"kind" binding:"required"`            // ui
-	Hash              string           `json:"hash"`                               // ui or bundler
-	SellToken         string           `json:"sellToken"`                          // optional for limit orders, ui
-	BuyToken          string           `json:"buyToken"`                           // ui
-	SellAmount        float64          `json:"sellAmount"`                         // optional for limit orders, ui
-	BuyAmount         float64          `json:"buyAmount"`                          // ui
-	PartiallyFillable bool             `json:"partiallyFillable"`                  // ui
-	CallData          string           `json:"callData"`                           // UI, Bundler, Solver
-	Status            ProcessingStatus `json:"status" binding:"status"`            // ui or bundler
-	CreatedAt         int64            `json:"createdAt" binding:"opt_int"`        // ui or bundler
-	ExpirationAt      int64            `json:"expirationAt" binding:"opt_int"`     // ui or bundler for default expiration (TTL: 100 seconds)
-	ChainID           *big.Int         `json:"chainId" binding:"required,chain_id"`
+	Sender            string           `json:"sender" binding:"required,eth_addr"`
+	From              Asset            `json:"From"`
+	To                Asset            `json:"To"`
+	PartiallyFillable bool             `json:"PartiallyFillable"`
+	Hash              string           `json:"hash"`
+	CallData          string           `json:"callData"`
+	Status            ProcessingStatus `json:"status" binding:"status"`
+	CreatedAt         int64            `json:"createdAt" binding:"opt_int"`
+	ExpirationAt      int64            `json:"expirationAt" binding:"opt_int"`
```

Comment:
Fields like PartiallyFillable & ExpirationAt  should be inside "extraData"

Reply:
we don;t need to maintain an new key:value